xtask/codegen/src/main.rs

```diff
@@ -27,124 +24,80 @@ use crate::generate_schema::generate_configuration_schema;
 #[cfg(feature = "website")]
 use crate::generate_website::generate_files;
 use crate::promote_rule::promote_rule;
+
 use generate_new_lintrule::*;
+use xtask::Mode::Overwrite;
 use xtask_codegen::{
-    generate_analyzer, generate_ast, generate_formatters, generate_parser_tests, generate_tables,
+    generate_analyzer, generate_ast, generate_crate, generate_formatters, generate_parser_tests,
+    generate_tables, task_command, TaskCommand,
 };
 
 fn main() -> Result<()> {
     let _d = pushd(project_root());
+    let result = task_command().run();
```

Comment:
There's no default command so when user runs it with no argument it will complain about missing command. While valid behavior - you can show help if no argument is specified for slightly better UX.

```suggestion
    let result = task_command().fallback_to_usage().run();
```

Reply:
I wanted to use `fallback_to_usage`, but it gave a compiler error. Turned out I forgot `bpaf(options)`

Reply:
Hmm... Interesting problem. I added `.run()` as shortcut for `.to_options().run()` to make life a bit easier, but turns out it can also add confusion. Will keep that in mind.